### PR TITLE
Linux testcases full plugins coverage

### PR DESCRIPTION
**Pull request by gcmoreira**
This PR adds the 10 missing test cases for complete full plugin coverage and also improve existing ones.

### CODE REVIEW

File: test/test_volatility.py [hunk 1]
```diff
@@ -640,16 +631,139 @@
...
+        assert rc == 0
+        assert out.count(b"\n") > 4
+
+        # PageVAddr PagePAddr MappingAddr .. DumpSafe
+        assert re.search(
+            rb"0xea000054c5f8\s0x18389000\s0x88001ab5c3b0.*?True",
+            out,
+        )
+        assert os.path.exists(inode_dump_filename)
+        inode_contents = open(inode_dump_filename, "rb").read()
```
[the remark this answers is not in the document]
Comment (gcmoreira) [reply test/test_volatility.py]: CodeQL you are wrong :)

[the remark this answers is not in the document]
Comment (ikelos) [reply test/test_volatility.py]: Can we please turn this into 
```
with open(inode_dump_filename, "rb") as fp:
    inode_contents = fp.read()
```
This ensures the file is closed after the read is finished (and should get rid of the CodeQL complaint too)

**Review by ikelos — COMMENTED** [review]
Happy to include this if you are.  Several of the tests just seem to be number of lines of output returned, but if that's enough to determine test success or failure then that's all good.  Just one fix required, opening the file as a `ContextManager` rather than just calling open please...

File: test/test_volatility.py [hunk 2]
```diff
@@ -640,16 +631,139 @@ def test_linux_page_cache_files(image, volatility, python):
...
```
Comment (ikelos) [inline test/test_volatility.py]:
```suggestion
        with open(inode_dump_filename, "rb") as fp:
            inode_contents = fp.read()
```

File: test/test_volatility.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ikelos) [reply test/test_volatility.py]: > CodeQL you are wrong :)

It doesn't like the garbage collector being used to close the file, it wants explicit code to do it, so technically... it's got a point?  5:S

[the remark this answers is not in the document]
Comment (gcmoreira) [reply test/test_volatility.py]: The file is indeed closed, but I'll explicitly close it so the poor AI can better understand it :)

[the remark this answers is not in the document]
Comment (ikelos) [reply test/test_volatility.py]: Thanks, you're too good for it, it doesn't deserve you...  5;P

**Comment by gcmoreira** [comment]
> Happy to include this if you are. Several of the tests just seem to be number of lines of output returned, but if that's enough to determine test success or failure then that's all good. Just one fix required, opening the file as a `ContextManager` rather than just calling open please...

Yes, that's enough to verify the plugin doesn't crash and that the requirements are met, which is already a significant step. Since the memory dump is quite old, some plugins don’t work with that kernel version. Even in these cases, I ensured that any crashes were due to the `unsupported version` and not another issue.

Additionally, some plugins only produce results when detecting something malicious. Since this sample doesn't trigger such findings, the best we can do is ensure the plugin doesn't crash, meets the required criteria, and displays the headers.

**Comment by gcmoreira** [comment]
That said, we should ensure that every time a new Linux plugin is added, at least a basic test case is included as part of the pull request. Some like this:

```python
def test_linux_<plugin_name>(image, volatility, python):
    rc, out, _err = runvol_plugin(
        "linux.<plugin_name>", image, volatility, python
    )
    assert rc == 0
    assert out.count(b"\n") >= 4
```

**Pull request merged**